### PR TITLE
🌱 Add github integration tests

### PR DESCRIPTION
**Pull request by nitisht**
**What this PR does / why we need it**:
This PR adds integration tests to test the release asset download flow from github

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squash commits
- [ ] include documentation
- [ ] add unit tests

### CODE REVIEW

**Comment by berendt** [comment]
We use Zuul CI. The use of Github Actions should be avoided. I would prefer not to build everything on Github Actions first and then migrate it all.

Benjamin recently wrote a Zuul users guide. It's available at https://docs.scs.community/dev-docs/operations/operations/zuul-ci-cd-quickstart-user-guide.

A good example for the use of Zuul is in the https://github.com/SovereignCloudStack/k8s-cluster-api-provider repository.

I think someone from the Zuul team can also give a hands-on.

There is also a dedicated Matrix room: #scs-zuul:matrix.org

File: Makefile [hunk 1]
```diff
@@ -325,6 +325,18 @@ test-unit: $(SETUP_ENVTEST) $(GOTESTSUM) $(HELM) ## Run unit
 	KUBEBUILDER_ASSETS="$(KUBEBUILDER_ASSETS)" $(GOTESTSUM) --junitfile=.coverage/junit.xml --format testname -- -mod=vendor \
 	-covermode=atomic -coverprofile=.coverage/cover.out -p=4 ./internal/controller/...
 
+.PHONY: test-unit-core
```
Comment (janiskemper) [inline Makefile:328]: there is test-unit already!

**Review by janiskemper — CHANGES_REQUESTED** [review]

**Comment by jschoone** [comment]
> We use Zuul CI. The use of Github Actions should be avoided. I would prefer not to build everything on Github Actions first and then migrate it all.
> 
> Benjamin recently wrote a Zuul users guide. It's available at https://docs.scs.community/dev-docs/operations/operations/zuul-ci-cd-quickstart-user-guide.
> 
> A good example for the use of Zuul is in the https://github.com/SovereignCloudStack/k8s-cluster-api-provider repository.
> 
> I think someone from the Zuul team can also give a hands-on.
> 
> There is also a dedicated Matrix room: #scs-zuul:matrix.org

Thanks for raising this. We are aware of it but agreed to use GitHub Actions here to move faster. 
A later use of another CI tool is not such a big hassle. Due to the container meeting not taking place, I forgot to report. If there is any ambiguity, it should be deferred to an appropriate meeting.

**Comment by janiskemper** [comment]
blocked by https://github.com/SovereignCloudStack/cluster-stack-operator/issues/20 and until there is a second release in [cluster-stacks](https://github.com/SovereignCloudStack/cluster-stacks) named docker-ferrol-1-27-v1

**Comment by nitisht** [comment]
> blocked by #20 and until there is a second release in [cluster-stacks](https://github.com/SovereignCloudStack/cluster-stacks) named docker-ferrol-1-27-v1

Now that #20 is closed, can we take this in?

**Review by janiskemper — APPROVED** [review]

**Pull request merged**